src/main/java/graphql/schema/GraphqlDirectivesContainerTypeBuilder.java

```diff
@@ -22,6 +21,33 @@ public B replaceAppliedDirectives(List<GraphQLAppliedDirective> directives) {
         return (B) this;
     }
 
+    public B addAppliedDirectives(GraphQLAppliedDirective... directives) {
+        assertNotNull(directives, () -> "directives can't be null");
+        for (GraphQLAppliedDirective directive : directives) {
+            addAppliedDirective(directive);
+        }
+        return (B) this;
+    }
+
+    public B addAppliedDirective(GraphQLAppliedDirective directive) {
+        assertNotNull(directive, () -> "directive can't be null");
+        this.appliedDirectives.add(directive);
+        return (B) this;
+    }
+
+    public B addAppliedDirective(GraphQLAppliedDirective.Builder builder) {
+        return addAppliedDirective(builder.build());
+    }
+
+    /**
+     * @param directives the variable args of directives
+     *
+     * @return this builder
+     *
+     * @deprecated - use {@link #replaceAppliedDirectives(List)} to clear and replace directives,
+     * or {@link #addAppliedDirectives(GraphQLAppliedDirective...)} to add directives without clearing the existing ones
+     */
+    @Deprecated(since = "2025-02-16", forRemoval = true)
```

Comment:
`forRemoval` is set to true because this behaviour is identical in `replaceAppliedDirectives`